crates/bevy_ecs/src/schedule/executor/multi_threaded.rs

```diff
@@ -473,7 +545,7 @@ impl MultiThreadedExecutor {
         &mut self,
         system_index: usize,
         _system: &BoxedSystem,
-        conditions: &mut Conditions,
+        conditions: &Conditions,
```

Comment:
Is this 100% required to be read-only?

Reply:
No, probably not.  

The issue is that those conditions come from the schedule, which is only available during `run`, so they couldn't get put in the `ExecutorState` behind the mutex.  Since they were ultimately boxed systems, I figured I could treat them like the other systems.  But, yeah, this is hard to prove safe.  We need *something* shared because every task needs a copy to run the executor with, but it can be better than this.  

Hmm, I could create a `Mutex<&mut Conditions>` for the duration of `run` and pass a shared reference to that alongside the other mutex, then lock one after the other.  Would that be cleaner?  That would also address your concerns about the SAFETY comments.